### ISSUE TITLE
when i use a vpn the transcipt fetching is not working also i cant run it in a ubuntu server

### ISSUE THREAD

**Issue by kanchana404** [Kakulukian/youtube-transcript#42]
when i use a vpn the transcipt fetching is not working also i cant run it in a ubuntu server

**Comment by hatemmezlini** [comment]
That is due to Youtube banning ISP ips.  I have made a working solution using oauth2 that doesn't uses proxies and avoids Youtube ban. You can use it for free on Apify: https://apify.com/invideoiq/video-transcript-scraper. You only pay for Apify usage, however Apify gives you free credit of 5$ which will give you around 5000 transcripts

**Comment by OdapX** [comment]
@hatemmezlini  what do you mean using oauth2 ? are you talking about the official captions api from google ? did it work for all publically available yt videos with your own credentials ?

**Comment by hatemmezlini** [comment]
@OdapX No, it's not the offical API. Initially it worked but I got banned using oauth2 using yt-dlp: https://github.com/yt-dlp/yt-dlp/wiki/Extractors#logging-in-with-oauth
 so I switched to another method

**Comment by OdapX** [comment]
@hatemmezlini  may I ask, what method you're referring to?

**Comment by cedricmenec** [comment]
Could you please indicate what error you're encountering?
For my part, when I use youtube-transcript locally, the transcript fetching works fine.

However, when I deploy my application on Vercel, I get the following error:

`Error: Failed to get YouTube video transcription: [YoutubeTranscript] Transcript is disabled on this video ({video_id})`

I suspect, without certainty, an issue related to YouTube banning Vercel IPs. Could you confirm if you get this type of error when you encounter an IP ban issue?

**Comment by kamikazebr** [comment]
> Could you please indicate what error you're encountering? For my part, when I use youtube-transcript locally, the transcript fetching works fine.
> 
> However, when I deploy my application on Vercel, I get the following error:
> 
> `Error: Failed to get YouTube video transcription: [YoutubeTranscript] Transcript is disabled on this video ({video_id})`
> 
> I suspect, without certainty, an issue related to YouTube banning Vercel IPs. Could you confirm if you get this type of error when you encounter an IP ban issue?

Did you put the code in backend? I think that should run in browser to use IP client, no?

**Comment by manasvimohan** [comment]
Same issue on netlify servers, not only vercel.
